packages/venia-concept/src/components/Checkout/form.js

```diff
@@ -1,329 +1,366 @@
-        submitShippingAddress: func.isRequired,
-        submitOrder: func.isRequired,
-        submitPaymentMethodAndBillingAddress: func.isRequired,
-        submitShippingMethod: func.isRequired,
-        submitting: bool.isRequired
-    };
-
-    /*
-     *  Class Properties.
-     */
-    get editableForm() {
-        const {
-            editing,
-            submitting,
-            isAddressIncorrect,
-            incorrectAddressMessage,
-            directory
-        } = this.props;
-        const { countries } = directory;
-
-        switch (editing) {
-            case 'address': {
-                const { shippingAddress } = this.props;
-
-                return (
-                    <AddressForm
-                        initialValues={shippingAddress}
-                        submitting={submitting}
-                        countries={countries}
-                        cancel={this.stopEditing}
-                        submit={this.submitShippingAddress}
-                        isAddressIncorrect={isAddressIncorrect}
-                        incorrectAddressMessage={incorrectAddressMessage}
-                    />
-                );
-            }
-            case 'paymentMethod': {
-                const { billingAddress } = this.props;
-
-                return (
-                    <PaymentsForm
-                        cancel={this.stopEditing}
-                        initialValues={billingAddress}
-                        submit={this.submitPaymentMethodAndBillingAddress}
-                        submitting={submitting}
-                        countries={countries}
-                    />
-                );
-            }
-            case 'shippingMethod': {
-                const { availableShippingMethods, shippingMethod } = this.props;
-
-                return (
-                    <ShippingForm
-                        availableShippingMethods={availableShippingMethods}
-                        cancel={this.stopEditing}
-                        shippingMethod={shippingMethod}
-                        submit={this.submitShippingMethod}
-                        submitting={submitting}
-                    />
-                );
-            }
-            default: {
-                return null;
-            }
-        }
-    }
+const isSubmitDisabled = (busy, valid) => busy || !valid;
+
+const EditableForm = props => {
+    const {
+        editing,
+        setEditing,
+        submitShippingAddress,
+        submitShippingMethod,
+        submitPaymentMethodAndBillingAddress,
+        submitting,
+        isAddressIncorrect,
+        incorrectAddressMessage,
+        directory: { countries }
+    } = props;
+
+    const handleCancel = useCallback(() => {
+        setEditing(null);
+    }, [setEditing]);
+
+    const handleSubmitAddressForm = useCallback(
+        async formValues => {
+            await submitShippingAddress({
+                formValues
+            });
+            setEditing(null);
```

Comment:
If we move state out of redux we must at some point `await` some async action completion where the logic would have [otherwise been handled internally in the reducer](https://github.com/magento-research/pwa-studio/blob/develop/packages/venia-concept/src/reducers/checkout.js#L116). For example, we would have `setEditing(null)` in the `paymentMethod.accept` function instead. We will see the same sort of logic when/if we convert the checkout `step` to a local state.

Is there some best practice when it comes to doing this sort of thing? Personally I think this reads better but it has a consequence of "hiding" these state changes within the components.

Reply:
After discussion with @jimbo we decided that while this may be the right direction to take when it comes to refactoring out redux, we don't want to do it _here_. So I will be reverting this commit into a separate branch/PR so we have it around (since I spent time on updating tests). And we can deal with it later.

Reply:
Moved to #1338